src/megaapi_impl.cpp

```diff
@@ -3572,68 +3604,130 @@ MegaStringMapPrivate::MegaStringMapPrivate(const MegaStringMapPrivate *megaStrin
     delete keys;
 }
 
-MegaStringListPrivate::MegaStringListPrivate()
+
+MegaStringListPrivate::MegaStringListPrivate(char **newlist, int size)
+{
+    for (int i = 0; i < size; i++)
+    {
+        append(newlist[i]);
```

Comment:
This function used to take ownership of the parameters but now copies them - are all the callers adjusted, and can we be sure this isn't introducing a memory leak?

Reply:
I actually thought this was a bug that it wouldn't copy. I will check...

Reply:
Reverted the changes to this class